openwisp_monitoring/monitoring/base/models.py

```diff
@@ -213,13 +213,18 @@ def write(
             database=database,
             retention_policy=retention_policy,
         )
-        timeseries_write.delay(name=self.key, values=values, **options)
-        post_metric_write.send(**signal_kwargs)
         # check can be disabled,
         # mostly for automated testing and debugging purposes
-        if not check:
-            return
-        self.check_threshold(value, time, retention_policy, send_alert)
+        if check:
+            options['threshold_dict'] = {
```

Comment:
`threshold_dict` should be named `check_threshold_kwargs`, which indicates these are the keyword arguments for the `check_threshold()` method.